src/entity/Customer.java

```diff
@@ -0,0 +1,62 @@
+package entity;
+
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+
+public class Customer extends Thread {
+    private long id;
+    private int position;
+    private Company company;
+
+    public static final Logger logger = Logger.getLogger(Customer.class.getName());
+
+    public Customer(int id) {
+        this.id = id;
+        this.company = Company.getCompany();
+    }
+
+    public long getId() {
+        return id;
+    }
+
+    public int getPosition() {
+        return position;
+    }
+
+    public void setPosition(int position) {
+        this.position = position;
+    }
+
+    public void callTaxi() {
+        logger.info("Client with ID " + getId() + " calls taxi!");
+
+        try {
+            for (Car car : company.getCars()) {
```

Comment:
Customer shouldn't have any logic to retrieve car of the company. Customer should place order only. Inside e.g. placeOrder method company can run receiveOrder method.